compiler/rustc_mir_transform/src/const_eval_select.rs

```diff
@@ -0,0 +1,40 @@
+//! Replaces calling `const_eval_select` with calling its runtime argument.
+
+use rustc_index::vec::Idx;
+use rustc_middle::{
+    mir::{Body, Field, MirPass, Operand, TerminatorKind},
+    ty::{self, TyCtxt},
+};
+
+pub struct ReplaceConstEvalSelect;
+
+impl<'tcx> MirPass<'tcx> for ReplaceConstEvalSelect {
+    fn run_pass(&self, tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+        let (basic_blocks, local_decls) = body.basic_blocks_and_local_decls_mut();
+        for block in basic_blocks.iter_mut() {
+            let terminator = block.terminator.as_mut().unwrap();
+            if let TerminatorKind::Call { func, args, .. } = &mut terminator.kind {
+                if let ty::FnDef(def_id, _) = func.ty(local_decls, tcx).kind() {
+                    if Some(*def_id) == tcx.lang_items().const_eval_select() {
```

Comment:
maybe...
❌ `if Some(*def_id) == tcx.lang_items().const_eval_select()`
⭕ `if intrinsic_name == sym::const_eval_select`

Reply:
So... one thing that could in theory happen (but doesn't happen here) is that we create a function pointer to `const_eval_select` calling something, and then it would get monomorphized.

From the backtrace it looks like we're codegenning the `at_compiletime` helper function. So... maybe this is happening because of the monomorphization collector. I was afraid of that ^^ let me grab you a link

Reply:
I misunderstood... These are equivalent...

Reply:
Ok... so... the collector will pick up the generics somewhere in the visitor in https://github.com/rust-lang/rust/blob/17dfae79bbc3dabe1427073086acf7f7bd45148c/compiler/rustc_monomorphize/src/collector.rs#L627

at this point... I think maybe we should give up XD I'm sorry for sending you on a wild goose chase. Let's just codegen the intrinsics to abort or sth and be done with it. Unfortunate, but it seems to fragile to continue here.

Reply:
I think both intrinsics could just abort, as we can use const eval select for the runtime behavior. Implementing const_alloc cannot just be a nop, only dealloc can. So we'd need some complex logic anyway